<?xml version="1.0"?>
<launch>
    <group ns="Vehicle/Lidar">
        <node name="LaserScansSender" pkg="tod_lidar" type="LaserScansSender" output="screen"/>
        <node name="ObjectListsSender" pkg="tod_lidar" type="ObjectListsSender" output="screen"/>

        <node name="LaserScansDetector" pkg="tod_lidar" type="LaserScansDetector" output="screen">
            <param name="min_nof_pts" value="3"/>
            <param name="max_nof_pts" value="30"/>
            <param name="max_distance" value="0.49"/>
            <param name="x_min" value="0.0"/>
            <param name="x_max" value="15.0"/>
            <param name="y_min_max" value="8.5"/>
        </node>

        <group ns="GridMap">
            <node name="Creator" pkg="tod_lidar" type="GridMapCreator" output="screen">
                <rosparam command="load" file="$(find tod_lidar)/config/grid_map.yaml" />
            </node>

            <node name="GridMapDetector" pkg="tod_lidar" type="GridMapDetector" output="screen"/>

            <node name="Visualizer" pkg="grid_map_visualization" type="grid_map_visualization" output="log">
                <rosparam command="load" file="$(find tod_lidar)/config/grid_map_helper.yaml"/>
                <remap from="Visualizer/elevation_grid" to="elevation_grid"/>
            </node>
        </group>
    </group>
</launch>
